client/src/lib/tests/attempt-result.test.js

```diff
@@ -295,6 +296,72 @@ describe('meetsCutoff', () => {
   });
 });
 
+describe('isWorldRecord', () => {
+  it('returns true when a new world record average is achieved', () => {
+    const attemptResult = 500;
+    const worldRecords = [
+      {
+        type: 'average',
+        event: {
+          id: '333',
+        },
+        attemptResult: 701,
+      },
+    ];
+    expect(isWorldRecord(attemptResult, '333', 'average', worldRecords)).toBe(
+      true
+    );
+  });
+
+  it('returns false when a new world record average is not achieved', () => {
+    const attemptResults = 500;
+    const worldRecords = [
+      {
+        type: 'average',
+        event: {
+          id: '333',
+        },
+        attemptResult: 200,
+      },
+    ];
+    expect(isWorldRecord(attemptResults, '333', 'average', worldRecords)).toBe(
+      false
+    );
+  });
+
+  it('returns true when a new world record single is achieved', () => {
+    const attemptResult = 500;
+    const worldRecords = [
+      {
+        type: 'single',
+        event: {
+          id: '333',
+        },
+        attemptResult: 701,
+      },
+    ];
+    expect(isWorldRecord(attemptResult, '333', 'single', worldRecords)).toBe(
+      true
+    );
+  });
+
+  it('returns false when a new world record single is not achieved', () => {
+    const attemptResults = 500;
+    const worldRecords = [
+      {
+        type: 'single',
+        event: {
+          id: '333',
+        },
+        attemptResult: 200,
+      },
+    ];
+    expect(isWorldRecord(attemptResults, '333', 'single', worldRecords)).toBe(
+      false
+    );
+  });
+});
```

Comment:
Let's add one more tests when there is no world record entry of the given type (MLBD average), in which case it should return `false`.

Reply:
Done!